backend/api/Controllers/DeckController.cs

```diff
@@ -81,14 +85,43 @@ public async Task<ActionResult<Deck>> GetDeckById([FromRoute] string id)
                 var deck = await _deckService.ReadById(id);
                 if (deck == null)
                     return NotFound($"Could not find deck with id {id}");
-                return Ok(deck);
+                return Ok(new DeckResponse(deck));
             }
             catch (Exception e)
             {
                 _logger.LogError(e, "Error during GET of deck from database");
                 throw;
             }
+        }
 
+        /// <summary>
+        /// Lookup all the mission definitions related to a deck
+        /// </summary>
+        [HttpGet]
+        [Authorize(Roles = Role.Any)]
+        [Route("{id}/mission-definitions")]
+        [ProducesResponseType(typeof(MissionDefinitionResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<IList<MissionDefinitionResponse>>> GetMissionDefinitionsInDeck([FromRoute] string id)
+        {
+            try
+            {
+                var deck = await _deckService.ReadById(id);
+                if (deck == null)
+                    return NotFound($"Could not find deck with id {id}");
+
+                var missionDefinitions = await _missionDefinitionService.ReadByDeckId(deck.Id);
+                var missionDefinitionResponses = missionDefinitions.FindAll(m => !m.IsDeprecated).Select(m => new CondensedMissionDefinitionResponse(m));
+                return Ok(missionDefinitionResponses);
```

Comment:
Is it correct to have a `CondensedMissionDefinitionResponse` here or should it be a `MissionDefinitionResponse`?

Reply:
The return type here is inconsistent but I think it's still good to have the condensed one since returning a list of the uncondensed ones is currently very slow. Each mission definition requires a blob storage lookup. If the frontend wants a more detailed mission definition they can then make a request for a specific mission definition. In the future I do think though that we may want to move away from storing custom mission definition sources in the blob store and instead store them directly in the SQL for the Source table, as we do with MissionRuns

Reply:
I've at least made them consistent now